test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ParsingSimpleTypeModelBinderTest.cs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    public class ParsingSimpleTypeModelBinderTest
+    {
+        public static TheoryData<Type> ConvertableTypeData
+        {
+            get
+            {
+                return new TheoryData<Type>
+                {
+                    typeof(decimal),
+                    typeof(double),
+                    typeof(float),
+                };
+            }
+        }
+
+        public static TheoryData<Type> NullableConvertableTypeData
+        {
+            get
+            {
+                return new TheoryData<Type>
+                {
+                    typeof(decimal?),
+                    typeof(double?),
+                    typeof(float?),
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ConvertableTypeData))]
+        [MemberData(nameof(NullableConvertableTypeData))]
+        public async Task BindModel_ReturnsFailure_IfTypeCanBeConverted_AndConversionFails(Type destinationType)
```

Comment:
FYI these names I think may have been copy pasted. The `IfTypeCanBeConverted` suggests an older iteration of the model binding design.